stellargraph/layer/graphsage.py

```diff
@@ -1048,56 +1086,70 @@ class DirectedGraphSAGE(GraphSAGE):
         kernel_regularizer (str or func, optional): The regulariser to use for the weights of each layer;
             defaults to None.
 
-    Note: If a generator is not specified, then additional keyword arguments must be supplied:
-        in_samples (list): The number of in-node samples per layer in the model.
-        out_samples (list): The number of out-node samples per layer in the model.
-        input_dim (int): The dimensions of the node features used as input to the model.
-    """
+    Note::
+        If a generator is not specified, then additional keyword arguments must be supplied:
+
+        * in_samples (list): The number of in-node samples per layer in the model.
+
+        * out_samples (list): The number of out-node samples per layer in the model.
+
+        * input_dim (int): The dimensions of the node features used as input to the model.
+
+        * multiplicity (int): The number of nodes to process at a time. This is 1 for a node inference 
+          and 2 for link inference (currently no others are supported).    """
 
     def _get_sizes_from_generator(self, generator):
         """
         Sets in_samples, out_samples and input_feature_size from the generator.
         Args:
              generator: The supplied generator.
         """
-        self.in_samples = generator.generator.in_samples
+
+        if not isinstance(generator, (DirectedGraphSAGENodeGenerator,)):
```

Comment:
Should this test include `DirectedGraphSAGELinkGenerator`?

Reply:
It currently can't include `DirectedGraphSAGELinkGenerator` as we don't have one.  We should get onto that sometime.